wpimath/src/main/java/edu/wpi/first/math/estimator/DifferentialDrivePoseEstimator.java

```diff
@@ -187,50 +193,37 @@ public Pose2d getEstimatedPosition() {
    *     or sync the epochs.
    */
   public void addVisionMeasurement(Pose2d visionRobotPoseMeters, double timestampSeconds) {
-    // Step 1: Get the pose odometry measured at the moment the vision measurement was made.
+    // Step 1: Get the odometry pose measured at the moment the vision measurement was made.
     var sample = m_poseBuffer.getSample(timestampSeconds);
 
-    if (sample.isEmpty()) {
+    if (sample.isEmpty()
+        || timestampSeconds < m_poseBuffer.getInternalBuffer().firstKey()
+        || timestampSeconds > m_poseBuffer.getInternalBuffer().lastKey()) {
       return;
     }
```

Comment:
Should this change be made in TimeInterpolatableBuffer instead? If this is intended to fix https://github.com/wpilibsuite/allwpilib/issues/4952, it should be in its own PR.